### PR TITLE
Pipelines - Wait for Workloads readiness or fail

### PR DESCRIPTION
**Pull request by mathieu-benoit**

### CODE REVIEW

File: pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    development: deployer
+
+# Because this pipeline runs tests, we're going to use concurrency controls to ensure only one deploy and test cycle is running at a time against the target environment.
+concurrency:
+  group: "${{ pipeline.id }}-${{ pipeline.app.id }}-${{ inputs.env_id }}"
+
+jobs:
+  deploy:
+    steps:
+    - if: ${{ ! inputs.set_id }}
```
Comment (astromechza) [inline pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml:20]: probably useful to indicate that these two steps are copied from teh default deployment pipeline

File: pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    - name: Wait for Workloads readiness
+      continue-on-error: true
+      id: wait-for-workloads-readiness
+      uses: actions/humanitec/wait-for-readiness@v1
+      with:
+        env_id: ${{ inputs.env_id }}
+
+    - if: ${{ steps.wait-for-workloads-readiness.status == 'failed' }}
+      name: Fail if Workloads readiness failed
```
Comment (astromechza) [inline pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml:44]: Can you add a comment explaining why this is broken out to a separate step from the previous one?

**Review by astromechza — COMMENTED** [review]
Looks great so far. 2 suggestions on more inline comments

File: pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+When the Platform Orchestrator deploys changes into an Environment, the Deployment status indicates whether it successfully provisioned associated resources and configured the Workloads. It does not necessarily imply that the Workloads are healthy or that the desired functionality is available to end users. The Pipeline will also be successfull in scenario where the Workload will never be `Ready`, for example when the container image doesn't exist (`ImagePullBackOff`), or when an admission controller is blocking the deployment in the Kubernetes cluster, etc.
+
+Humanitec Pipelines can be used to wait for the readiness of the Worklaods deployed in Kubernetes, and fail the Pipeline if the previous step timed out.
+
+This Pipeline definition shows how to accomplish this.
+
+Note: the `Deployment status` on the Deployment in Humanitec will still be `Successful`.
```
Comment (TobiasBabin) [inline pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/README.md:7]: Name the concrete product whenever possible.
```suggestion
Note: the `Deployment status` on the Deployment in the Platform Orchestrator will still be `Successful`.

```

File: pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml (hunk 2 above)
Comment (mathieu-benoit) [reply pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml:51]: I added something, but just to make sure, why do we need this again? Failure of the previous step (with timeout) won't be enough by itself?

Comment (astromechza) [reply pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml:51]: @mathieu-benoit It should be fine by itself, the extra step is just to add a custom message or some slack notification or something.

**Review by astromechza — APPROVED** [review]
Ok looks good.

File: pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml (hunk 2 above)
Comment (mathieu-benoit) [reply pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml:51]: Gotcha, I'll update the comment on top of this step to clarify this. Thanks.

Comment (mathieu-benoit) [reply pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml:51]: Done.

**Review by TobiasBabin — APPROVED** [review]
LGTM - thanks a lot!

**Pull request merged**